engine/src/main/java/org/terasology/rendering/world/WorldRenderer.java

```diff
@@ -23,60 +23,227 @@
 
+    /**
+     * This method triggers the execution of the rendering pipeline and, eventually, sends the output to the display
+     * or to a file, when grabbing a screenshot.
+     *
+     * @param renderingStage "MONO" for standard rendering and "LEFT_EYE" or "RIGHT_EYE" for stereoscopic displays.
+     */
     void render(WorldRenderingStage renderingStage);
 
+    /**
+     * Gives the chance to an implementation to dispose any internal reference that might need a more careful disposal.
+     */
     void dispose();
 
+    /**
+     * Used only during the early part of the WorldRenderer lifecycle, this method triggers the loading of
+     * chunks around the camera and the generation of their meshes.
+     *
+     * @return Returns True if all necessary chunks have been loaded and their meshes have been generated, False otherwise.
+     */
+    // TODO: it might be desirable to separate the loading of chunks and mesh generation. The former may concern
+    // TODO: systems dealing with beyond-the-horizon simulations. The latter usually only concerns the renderer.
     boolean pregenerateChunks();
 
+    /**
+     * Returns a reference to the WorldProvider in use by the WorldRenderer.
+     *
+     * @return a WorldProvider instance
+     */
+    // TODO: candidate for removal: nothing uses it and can be obtained through the context
     WorldProvider getWorldProvider();
 
+    /**
+     * Provided a viewDistance value this method triggers the change in how far
+     * from the camera chunks are kept in memory and displayed.
+     *
+     * @param viewDistance a new viewDistance value.
+     */
+    // TODO: rename to setViewDistance - ensure it does nothing if the viewDistance hasn't changed.
     void changeViewDistance(ViewDistance viewDistance);
 
+    /**
+     * Returns the intensity of light at the current camera position due to the main light source (sun or moon).
+     *
+     * @return a float value
+     */
+    // TODO: candidate for removal - use getSunlightValueAt instead.
     float getSunlightValue();
 
+    /**
+     * Returns the intensity of the light at the current camera position due to sources in the scene, i.e. torches,
+     *
+     * @return a float value
+     */
+    // TODO: candidate for removal - use getBlocklightValueAt instead.
     float getBlockLightValue();
 
-    float getRenderingLightValueAt(Vector3f vector3f);
-
-    float getSunlightValueAt(Vector3f worldPos);
-
-    float getBlockLightValueAt(Vector3f worldPos);
-
+    /**
+     * Returns the intensity of the light at a given location due to the combination of main light (sun or moon)
+     * and in-scene light sources, i.e. torches.
+     *
+     * @param coordinates a Vector3f providing the coordinates of the block to calculate the light intensity from.
+     * @return a float value representing the intensity of the main light and in-scene lights combined
+     */
+    float getRenderingLightValueAt(Vector3f coordinates);
```

Comment:
Somebody should check my interpretation of what the code in WorldRendererImpl does against this method's javadoc.